File: app-java/app/src/main/java/com/phunware/android/phunwareproducthomework/features/detail/ZipCodeDetailViewModel.java
package com.phunware.android.phunwareproducthomework.features.detail;

import com.phunware.android.weathersdk.WeatherSDK;
import com.phunware.android.weathersdk.models.Weather;
import com.phunware.android.weathersdk.util.WeatherLiveData;

import androidx.lifecycle.ViewModel;

public class ZipCodeDetailViewModel extends ViewModel {
    private String zipCode;

    public WeatherLiveData<Weather> getWeather(String zipCode) {
        this.zipCode = zipCode;

        return WeatherSDK.getInstance().getWeather(zipCode);
    }
}

File: app-kotlin/app/src/main/java/com/phunware/android/phunwareproducthomework/features/detail/ZipCodeDetailViewModel.kt
package com.phunware.android.phunwareproducthomework.features.detail

import androidx.lifecycle.ViewModel
import com.phunware.android.weathersdk.util.WeatherLiveData
import com.phunware.android.weathersdk.WeatherSDK
import com.phunware.android.weathersdk.models.Weather

class ZipCodeDetailViewModel : ViewModel() {
    private lateinit var zipCode: String

    private val liveData: WeatherLiveData<Weather> by lazy {
        WeatherSDK.getWeather(zipCode)
    }

    fun getWeather(zipCode: String): WeatherLiveData<Weather> {
        this.zipCode = zipCode

        return liveData
    }
}